numba/pycc/modulemixin.c

```diff
@@ -112,6 +116,10 @@ PYCC(pycc_init_) (PyObject *module, PyMethodDef *defs,
                                     env_def_t *envs,
                                     env_gv_t *envgvs)
 {
+    _numba_hashsecret_siphash_k0 = _Py_HashSecret.siphash.k0;
+    _numba_hashsecret_siphash_k1 = _Py_HashSecret.siphash.k1;
+    _numba_hashsecret_djbx33a_suffix = _Py_HashSecret.djbx33a.suffix;
+
```

Comment:
What I think this is doing is a) providing a symbol, which is what the original issue was and b) making the values for the hash secret correct for the current python process. If you immediately import the AOT compiled module as in https://github.com/numba/numba/issues/6386#issuecomment-724043361 these end up being valid as a result of doing this assignment. If you import from another python process then they get over written with its values when the symbol map injection occurs via:
https://github.com/numba/numba/blob/f82ed9ff82f31a1ab87dc1511a442cfc2d4a22fa/numba/cpython/hashing.py#L667-L671

Reply:
I think it's probably worth adding a description along these lines to comments in the code around here and/or where `_numba_hashsecret_siphash_k0` etc. are declared - it will be kind to a future reader to give them the explanation with the code, rather than making them dig through the original issue to try and determine what this solves.

Reply:
added comment here: 982e225